packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -1058,6 +1057,8 @@ export class DeltaManager
         this._outbound.clear();
         this.emit("disconnect", reason);
 
+        // Avoid re-entrncy - remove all listeners before closing!
+        connection.removeAllListeners();
```

Comment:
Nit: I mentioned elsewhere that I'm not a fan of `removeAllListeners()`, but it's probably benign here since I don't think we hand out the connection object.

Reply:
Correct, I think code would be more convoluted if we keep remembering callbacks and unregister them one by one - more chanced to screw it up - all of this logic is already too complex to easily review and spot bugs :)